src/main/java/com/unstoppabledomains/resolution/naming/service/UNS.java

```diff
@@ -23,23 +23,32 @@
 import java.util.List;
 import java.util.Map;
 
-public class CNS extends BaseNamingService {
+public class UNS extends BaseNamingService {
   private final ProxyReader proxyReaderContract;
   
-  public CNS(NSConfig config, IProvider provider) {
+  public UNS(NSConfig config, IProvider provider) {
     super(config, provider);
     String proxyReaderAddress = NetworkConfigLoader.getContractAddress(config.getChainId(), "ProxyReader");
     this.proxyReaderContract = new ProxyReader(config.getBlockchainProviderUrl(), proxyReaderAddress, provider);
   }
 
   @Override
   public NamingServiceType getType() {
-    return NamingServiceType.CNS;
+    return NamingServiceType.UNS;
   }
 
-  public Boolean isSupported(String domain) {
+  public Boolean isSupported(String domain) throws NamingServiceException {
     String[] split = domain.split("\\.");
-    return (split.length != 0 && !split[split.length - 1].equals("zil"));
+    if (split.length == 0 || split[split.length - 1].equals("zil")) {
+      return false;
+    }
+    BigInteger tokenID;
+    try {
+      tokenID = tokenID(split[split.length - 1]);
+    } catch (Exception e) {
```

Comment:
I don't think it's a good idea to catch any exception. Please, specify what types of exceptions could mean that domain is not supported